File: articles/institutional-resilience/go/resilience_index.go
package main

import "fmt"

type Institution struct {
	Robustness               float64
	AdaptiveCapacity         float64
	RecoveryCapacity         float64
	TransformationalCapacity float64
	Legitimacy               float64
	Trust                    float64
	FeedbackQuality          float64
	LearningRate             float64
	Redundancy               float64
	Modularity               float64
	Coordination             float64
	ShockIntensity           float64
}

func ResilienceIndex(x Institution) float64 {
	return 0.10*x.Robustness +
		0.12*x.AdaptiveCapacity +
		0.10*x.RecoveryCapacity +
		0.08*x.TransformationalCapacity +
		0.12*x.Legitimacy +
		0.10*x.Trust +
		0.10*x.FeedbackQuality +
		0.08*x.LearningRate +
		0.07*x.Redundancy +
		0.05*x.Modularity +
		0.08*x.Coordination -
		0.10*x.ShockIntensity
}

func main() {
	demo := Institution{80, 75, 70, 60, 85, 82, 78, 74, 65, 68, 80, 50}
	fmt.Printf("Institutional resilience raw score: %.2f\n", ResilienceIndex(demo))
}

File: articles/institutional-resilience/rust/resilience_index.rs
struct Institution {
    robustness: f64,
    adaptive_capacity: f64,
    recovery_capacity: f64,
    transformational_capacity: f64,
    legitimacy: f64,
    trust: f64,
    feedback_quality: f64,
    learning_rate: f64,
    redundancy: f64,
    modularity: f64,
    coordination: f64,
    shock_intensity: f64,
}

fn resilience_index(x: &Institution) -> f64 {
    0.10 * x.robustness
        + 0.12 * x.adaptive_capacity
        + 0.10 * x.recovery_capacity
        + 0.08 * x.transformational_capacity
        + 0.12 * x.legitimacy
        + 0.10 * x.trust
        + 0.10 * x.feedback_quality
        + 0.08 * x.learning_rate
        + 0.07 * x.redundancy
        + 0.05 * x.modularity
        + 0.08 * x.coordination
        - 0.10 * x.shock_intensity
}

fn main() {
    let demo = Institution {
        robustness: 80.0,
        adaptive_capacity: 75.0,
        recovery_capacity: 70.0,
        transformational_capacity: 60.0,
        legitimacy: 85.0,
        trust: 82.0,
        feedback_quality: 78.0,
        learning_rate: 74.0,
        redundancy: 65.0,
        modularity: 68.0,
        coordination: 80.0,
        shock_intensity: 50.0,
    };

    println!(
        "Institutional resilience raw score: {:.2}",
        resilience_index(&demo)
    );
}
